src/event-log/event-log-level.ts

```diff
@@ -0,0 +1,80 @@
+import type { ULID } from 'ulid';
+import type { Event, EventLog } from './event-log.js';
+
+import { base32crockford } from '@scure/base';
+import { Encoder } from '../utils/encoder.js';
+import { monotonicFactory } from 'ulid';
+import { createLevelDatabase, LevelWrapper } from '../store/level-wrapper.js';
+
+
+type EventLogLevelConfig = {
+ /**
+   * must be a directory path (relative or absolute) where
+   *  LevelDB will store its files, or in browsers, the name of the
+   * {@link https://developer.mozilla.org/en-US/docs/Web/API/IDBDatabase IDBDatabase} to be opened.
+  */
+  location: string,
+  createLevelDatabase?: typeof createLevelDatabase,
+};
+
+export class EventLogLevel implements EventLog {
+  config: EventLogLevelConfig;
+  level: LevelWrapper<string>;
+  ulid: ULID;
+
+  constructor(config: EventLogLevelConfig = { location: 'EVENTLOG' }) {
+    this.config = {
+      createLevelDatabase,
+      ...config,
+    };
+
+    this.level = new LevelWrapper<string>({ ...this.config, valueEncoding: 'utf8' });
+    this.ulid = monotonicFactory();
+  }
+
+  async open(): Promise<void> {
+    // `db.open()` is automatically called by the database constructor.  We're calling it explicitly
+    // in order to explicitly catch an error that would otherwise not surface until another method
+    // like `db.get()` is called.  Once `db.open()` has then been called, any read & write
+    // operations will again be queued internally until opening has finished.
+    return this.level.open();
+  }
+
+  close(): Promise<void> {
+    return this.level.close();
+  }
+
+  clear(): Promise<void> {
+    return this.level.clear();
+  }
+
+  async append(tenant: string, messageCid: string): Promise<string> {
+    const hashedTenant = this.hashTenant(tenant);
+    const tenantEventLog = await this.level.partition(hashedTenant);
+
+    const watermark = this.ulid();
+    await tenantEventLog.put(watermark, messageCid);
+
+    return watermark;
+  }
+  async getEventsAfter(tenant: string, watermark?: string | undefined): Promise<Event[]> {
+    const hashedTenant = this.hashTenant(tenant);
+    const tenantEventLog = await this.level.partition(hashedTenant);
+
+    const opts = watermark ? { gt: watermark } : {};
+    const events: Array<Event> = [];
+
+    for await (const [key, value] of tenantEventLog.iterator(opts)) {
+      const event = { watermark: key, messageCid: value };
+      events.push(event);
+    }
+
+    return events;
+  }
+
+  hashTenant(tenant: string): string {
+    const tenantBytes = Encoder.stringToBytes(tenant);
+
+    return base32crockford.encode(tenantBytes);
```

Comment:
why do we need to hash the `tenant`?  isn't it already a DID/CID?

if this is desired/necessary, should we be doing this for all the other places too (e.g. `MessageStoreLevel`)?

Reply:
`hash` isn't even the correct terminology here so apologies for that. currently i'm encoding only. i'll update the method name to `encode`. but to answer your question, i had two specific reasons in mind:
1. DIDs can be very long and in general can be different lengths (ion short form vs. long form). `base32crockford` produces a fixed-length output, which makes it easier to handle and compare than the variable-length
2. `base32crockford` encoding also produces output that consists of a limited set of alphanumeric characters, making it more suitable for use in contexts where special characters or binary data are not allowed or need to be avoided. Specifically in the case of `level`, you figured out that only ascii characters are allowed when creating sub-levels.

Reply:
Sharing similar sentiment as @dcrousso here.  Since we are dealing with DIDs here, this doesn't feel necessary. Assuming `base32crockford` really "just" encodes, it is only going to make the key (a lot) longer.

Reply:
IMO this should be a separate commit and should really be applied to _all_ places we use `tenant` with `LevelWrapper`

> 1. DIDs can be very long and in general can be different lengths (ion short form vs. long form). `base32crockford` produces a fixed-length output, which makes it easier to handle and compare than the variable-length

are you sure that `base32crockford` produces a fixed-length output?  that's not what im seeing :(
```
base32crockford.encode((new TextEncoder).encode('123'))    // 64S36
base32crockford.encode((new TextEncoder).encode('123456')) // 64S36D1N6R
```

> 2. `base32crockford` encoding also produces output that consists of a limited set of alphanumeric characters, making it more suitable for use in contexts where special characters or binary data are not allowed or need to be avoided. Specifically in the case of `level`, you figured out that only ascii characters are allowed when creating sub-levels.

i agree with the sentiment here, but aren't we already "safe" since it's a DID?  i.e. all characters should already be between `37` and `127`